### PR TITLE
feat: project as a Boundary integ test and GetDataset route

### PR DESCRIPTION
**Pull request by kpark277**
Issue #, if available:

Description of changes:
* Add happy path integ test for project as a boundary
* Add GetDataset route for ITAdmin only to be used for integ tests
* Update ITAdmin Dataset permissions so that they can only GetDataset

Checklist:

* [x] Have you successfully deployed to an AWS account with your changes?
* [x] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: solutions/swb-reference/src/dynamicRouteConfig.ts [hunk 1]
```diff
@@ -177,14 +188,12 @@ export const dynamicRoutesMap: DynamicRoutesMap = {
         action: 'READ',
         subject: {
           subjectType: SwbAuthZSubject.SWB_DATASET,
-          subjectId: '${datasetId}',
-          projectId: '${projectId}'
```
Comment (kpark277) [inline solutions/swb-reference/src/dynamicRouteConfig.ts:181]: This shouldn't have gotten deleted. Will add back in next PR.

File: solutions/swb-reference/integration-tests/tests/multiStep/paab.test.ts [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+    expect(listProjects2.data.filter((proj) => proj.id === project1Id).length).toEqual(0);
+    // Get Projects
+    try {
+      await pa2Session.resources.projects.project(project1Id).get();
+    } catch (err) {
+      checkHttpError(err, unauthorizedHttpError);
+    }
+
+    console.log('Verifying PA2 CANNOT see Dataset1...');
+    // List Datasets for Project
```
Comment (FernandoAranda) [inline solutions/swb-reference/integration-tests/tests/multiStep/paab.test.ts:156]: are we waiting for other implementation to uncomment this?

Comment (kpark277) [reply solutions/swb-reference/integration-tests/tests/multiStep/paab.test.ts:156]: Yeah, @bhudnell is currently working on List Datasets for Project.

**Review by FernandoAranda — APPROVED** [review]
